plugins/BEdita/Core/src/Configure/Engine/DatabaseConfig.php

```diff
@@ -57,14 +65,33 @@ public function __construct($applicationId = null)
     }
 
     /**
-     * Read from DB (or cache) $key group of paramenters (see database `config.context`)
-     * and return the results as an array.
+     * Read `$key` parameters group from database using cache.
+     *
+     * @param string|null $key The group of parameters to read from database (see `config.context`).
+     * @return array Parsed configuration values.
+     */
+    public function read($key): array
+    {
+        $cacheKey = sprintf('db_conf_%s_%d', $key, $this->applicationId);
+
+        return (array)Cache::remember(
```

Comment:
Maybe it should be better to clear cache when a configuration is saved/deleted in `ConfigTable` via `afterSave()` and `afterDelete()` and take adavantage of `cache()` method of `Query` object, for example

```php
$query = $config->find()->select(['name', 'context', 'content']);
$query->where(function (QueryExpression $exp) {
    return $exp->notIn('name', $this->reservedKeys);
});
if ($this->applicationId) {
    $query->andWhere(['application_id' => $this->applicationId]);
} else {
    $query->andWhere(['application_id IS' => null]);
}
if (!empty($key)) {
    $query->andWhere(['context' => $key]);
}

$cacheKey = sprintf('db_conf_%s_%d', $key, $this->applicationId);
$query->cache($key, self::CACHE_CONFIG);
```